canonical/metacontroller-rock
# Update rock to `v3.0.0` and add tests

closes #8 

# Manual testing
* clone this repo and checkout to the pr's branch
* build the rock with `tox -e pack`
* export the rock to docker with `tox -e export-to-docker`
* inject the image into the microk8s image cache
```
docker save metacontroller:3.0.0 > mymeta.tar
microk8s ctr image import mymeta.tar
```
* deploy this bundle.yaml with trust on top of juju and uk8s (with metacontroller using the rock):
```
bundle: kubernetes
name: test
applications:
  admission-webhook: { charm: ch:admission-webhook, channel: latest/edge, scale: 1, trust: true }
  metacontroller-operator: { charm: ch:metacontroller-operator, channel: latest/edge, scale: 1, trust: true, options: { metacontroller-image: docker.io/library/metacontroller:3.0.0 } }
  minio:                   { charm: ch:minio, channel: latest/edge,       scale: 1 }
  kfp-profile-controller:  { charm: ch:kfp-profile-controller, channel: latest/edge, scale: 1, trust: true }
  kubeflow-profiles:  { charm: ch:kubeflow-profiles, channel: latest/edge, scale: 1, trust: true }
relations:
- [kfp-profile-controller:object-storage, minio:object-storage]
```
* check the `DecoratorController` object was created
```
kubectl get DecoratorController -n <namespace>
```
you should see:
```
NAME                                    AGE
kubeflow-pipelines-profile-controller   8m2s
```
* create a profile
```
kubectl apply -f - <<EOF
apiVersion: kubeflow.org/v1
kind: Profile
metadata:
  name: profilename
spec:
  owner:
    kind: User
    name: userid2@email.com
EOF
```
* make sure the desired resources are created in the profile's namespace, find a list of the resources [here](https://github.com/canonical/kfp-operators/blob/main/charms/kfp-profile-controller/tests/integration/test_charm.py#L208-L216)
